docs/photutils/psf.rst

```diff
@@ -28,10 +28,379 @@ model-fitting analysis, regardless to exactly what kind of model is actually
+    ...                                      psf_model=my_psf_model,
+    ...                                      fitter=my_fitter, niters=1,
+    ...                                      fitshape=(7,7))
+    >>> # get photometry results
+    >>> photometry_results, residual_image = my_photometry(image=my_image)
+
+Where ``my_finder``, ``my_group``, and ``my_bkg`` may be any suitable callable
+function. This approach allows one to customize every part of the photometry
+process provided that their input/output are compatible with the input/ouput
+expected by `~photutils.psf.DAOPhotPSFPhotometry` . See the API documentation
+for details.
+
+Performing PSF Photometry
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's take a look in a simple example with simulated stars whose PSF is
+assumed to be Gaussian.
+
+First let's create an image with four overlapping stars:
+    
+.. plot::
+    :include-source:
+    
+    from astropy.table import Table 
+    from photutils.datasets import make_random_gaussians, make_noise_image, make_gaussian_sources
+
+    sigma_psf = 2.0
+    sources = Table()
+    sources['flux'] = [700, 800, 700, 800]
+    sources['x_mean'] = [12, 17, 12, 17]
+    sources['y_mean'] = [15, 15, 20, 20]
+    sources['x_stddev'] = sigma_psf*np.ones(4)
+    sources['y_stddev'] = sources['x_stddev']
+    sources['theta'] = [0, 0, 0, 0]
+    sources['id'] = [1, 2, 3, 4]
+    tshape = (32, 32)
+    image = (make_gaussian_sources(tshape, sources) +
+             make_noise_image(tshape, type='poisson', mean=6.,
+                              random_state=1) +
+             make_noise_image(tshape, type='gaussian', mean=0.,
+                              stddev=2., random_state=1))
+
+    from matplotlib import rcParams
+    import matplotlib.pyplot as plt
+    rcParams['image.cmap'] = 'viridis'
+    rcParams['image.aspect'] = 1  # to get images with square pixels
+    rcParams['image.interpolation'] = 'nearest'
+    rcParams['image.origin'] = 'lower'
+    rcParams['font.size'] = 16
+    
+    plt.imshow(image)
+    plt.title('Simulated data')
+    plt.colorbar(orientation='horizontal', fraction=0.046, pad=0.04)
+
+
+Then let's import the required parts to set up a `~photutils.psf.DAOPhotPSFPhotometry` object::
+
+    >>> from photutils.detection import IRAFStarFinder
+    >>> from photutils.psf import IntegratedGaussianPRF, DAOGroup
+    >>> from photutils.background import MMMBackground, MADStdBackgroundRMS
+    >>> from astropy.modeling.fitting import LevMarLSQFitter
+    >>> from astropy.stats import gaussian_sigma_to_fwhm
+
+Let's then instantiate the objects::
+
+    >>> bkgrms = MADStdBackgroundRMS()
+    >>> std = bkgrms(image) # doctest: +SKIP
+    >>> iraffind = IRAFStarFinder(threshold=3.5*std,
+    ...                           fwhm=sigma_psf*gaussian_sigma_to_fwhm,
+    ...                           minsep_fwhm=0.01, roundhi=5.0, roundlo=-5.0,
+    ...                           sharplo=0.0, sharphi=2.0) # doctest: +SKIP
```

Comment:
@hamogu this skip is needed because ``image`` is defined outside in the ``plot`` directive right above (and the variable ``std`` depends on ``image``). Should I repeat ``image`` definition here and get rid of the doctest skips? @bsipocz